CONTRIBUTING.md

```diff
@@ -29,3 +29,139 @@ Recommendations and requirements for how to best contribute to Parquet. We striv
+   [parquet-rs](https://github.com/apache/arrow-rs), however at the discretion
+   of the PMC any open source Parquet implementation may be acceptable.
+   Implementations whose contributors actively participate in the community
+   (e.g. keep their feature matrix up-to-date on parquet-site) are more likely
+   to be considered. If discussed as a requirement in step one, demonstration
+   of integration with a query engine is also required for this step. The
+   implementations must be made available publicly, and they should
+   be fit for inclusion (for example, they were submitted as a pull request
+   against the target repository and committers gave positive reviews).
+
+Unless otherwise discussed, it is expected the implementations will be developed
+from their respective main branch (i.e. backporting is not expected).
+
+3. After the first two steps are complete a formal vote is held on the Parquet
+   mailing list to officially ratify the feature.  After the vote passes the
+   format change is merged into the parquet-format repository and it is expected
+   the changes from step 2 will also be merged soon after. Before merging into
+   Parquet-java a parquet-format release must be performed.
+
+#### General guidelines/preferences on additions.
+
+1. To the greatest extent possible changes should have an option for forwards
+   compatibility (old readers can still read files).
+
+2. New encodings should be fully specified in this repository and ideally not
+   rely on an external dependencies for implementation (i.e. `parquet-format` is
+   the source of truth for the encoding).
+
+3. New compression mechanisms must have a pure Java implementation that can be
+   used as dependency in parquet-java.
+
+### Releases
+
+The Parquet PMC aims to do releases of the format package only as needed when
+new features are introduced. If multiple new features are being proposed
+simultaneously some features might be consolidated into the same release.
+Guidance is provided below on when implementations should enable features added
+to the specification.  Due to confusion in the past over Parquet versioning it
+is not expected that there will be a 3.x release of the specification in the
+foreseeable future.
+
+### Compatibility and Feature Enablement
+
+For the purposes of this discussion we classify features into the following buckets:
+
+1. Backwards compatible. A file written under an older version of the format
+   should be readable under a newer version of the format.
+
+2. Forwards compatible. A file written under a newer version of the format with
+   the feature enabled can be read under an older version of the format, but
+   some information might be missing or performance might be suboptimal.
+
+3. Forwards incompatible. A file written under a newer version of the format with
+   the feature enabled cannot be read under an older version of the format (e.g.
+   adding and using a new compression algorithm).
```

Comment:
One clarifying question here - does "cannot be read under an older version of the format" allow for "cannot be read _correctly_ under an older version of the format"? I suspect we want that be considered a breaking change.

For reference I have the Arrow spec in mind which [defines](https://arrow.apache.org/docs/format/Versioning.html#forward-compatibility) forward compatibility as "An older client library must be able to either read data generated from a new client library or **detect that it cannot properly read the data.**" (emphasis mine). Is the "detect" case what we're calling "Forwards incompatible" here?

Reply:
I tried to clarify this.  Forward incompatible features I think should provide a signal to old readers so ideally an appropriate error message can be returned.